src/main/java/seedu/task/model/TaskBook.java

```diff
@@ -277,6 +299,26 @@ public void clearAllTasks(){
 		List<Integer> clearedTasksIndices = new ArrayList<Integer>();
 		List<String> clearedStatus = new ArrayList<String>();
 		
+		prepareClearedTasksForUndo(copyTasks, clearedTasks, clearedTasksIndices, clearedStatus);
+		
+		//actually remove the completed tasks
+		for (Task readTask : copyTasks) {
+			try {			
+				tasks.remove(readTask);
+			} catch (TaskNotFoundException e) {
+				assert false : "The target task cannot be missing";
+			}
+		}
+		
+		undoTaskStack.pushClearAllToUndoStack(clearedTasks, clearedTasksIndices, clearedStatus, UNDO_CLEAR_ALL_COMMAND);
+	}
+
+	/*
```

Comment:
Javadocs comment format is /** */.  You can remove the comment on line 322 since you already have it as a header comment